memory-viz/bin/cli.js

```diff
@@ -7,26 +7,31 @@ const { program } = require("commander");
 const { json } = require("node:stream/consumers");
 
 function parseFilePath(input) {
-    if (input) {
-        return pathExists(input, `File`);
-    } else {
-        return undefined;
+    const fullPath = path.resolve(process.cwd(), input);
+    if (fs.existsSync(fullPath, "File")) {
+        return fullPath;
     }
+    console.error(`Error: File ${fullPath} does not exist.`);
+    process.exit(1);
 }
 
 function parseOutputPath(input) {
-    return pathExists(input, `Output path`);
-}
-
-// helper function for parsing paths
-function pathExists(inputPath, errMsg) {
-    const fullPath = path.resolve(process.cwd(), inputPath);
-    if (!fs.existsSync(fullPath)) {
-        console.error(`Error: ${errMsg} ${fullPath} does not exist.`);
+    if (isPathDirectory(input)) {
+        console.error(`Error: Output path ${input} must be a file.`);
         process.exit(1);
-    } else {
-        return fullPath;
     }
+    const parsedPath = path.parse(input);
+    const outputDir = parsedPath.dir || ".";
+    const filename = `${parsedPath.name}.svg`;
```

Comment:
Use `parsedPath.base` to preserve the full filename the user inputs; we don't want to force a specific extension that overwrites whatever the user put in.